app/rest/metrics-collector/src/main/java/io/syndesis/rest/metrics/collector/PodMetricsReader.java

```diff
@@ -0,0 +1,274 @@
+        if (found != null) {
+
+            String pattern = String.format("%s:context=%s,type=routes,*", found.getDomain(), found.getKeyProperty("context"));
+            J4pSearchResponse sr = jolokia.execute(new J4pSearchRequest(pattern));
+
+            List<J4pReadRequest> list = new ArrayList<J4pReadRequest>();
+            for (ObjectName on : sr.getObjectNames()) {
+                list.add(new J4pReadRequest(on, ROUTE_ID, RESET_TIMESTAMP, EXCHANGES_TOTAL, EXCHANGES_FAILED, LAST_COMPLETED_TIMESTAMP, LAST_FAILED_TIMESTAMP, START_TIMESTAMP ));
+            }
+
+            List<J4pReadResponse> lrr = jolokia.execute(list);
+            for (J4pReadResponse rr : lrr) {
+                String routeId = rr.getValue(ROUTE_ID).toString();
+                if (filter == null || routeId.matches(filter)) {
+                    Map<String, String> row = new LinkedHashMap<String, String>();
+                    for (String attribute : rr.getAttributes()) {
+                        if (rr.getValue(attribute) != null) {
+                            row.put(attribute, rr.getValue(attribute).toString());
+                        }
+                    }
+                    answer.add(row);
+                }
+            }
+
+        }
+
+        // sort the list
+        Collections.sort(answer, new Comparator<Map<String, String>>() {
+            @Override
+            public int compare(Map<String, String> o1, Map<String, String> o2) {
+                // group by camel context first, then by route name
+                String c1 = o1.get("camelContextName");
+                String c2 = o2.get("camelContextName");
+                int answer = c1.compareTo(c2);
+                if (answer == 0) {
+                    // okay from same camel context, then sort by route id
+                    answer = o1.get("routeId").compareTo(o2.get("routeId"));
+                }
+                return answer;
+            }
+        });
+        return answer;
+    }
+
+    /**
+     * End of Camel Controller Code.
+     */
+
+
+    private enum TimeUnits {
+        hours(3600),
+        minutes(60),
+        seconds(1);
+
+        private final long duration;
+
+        TimeUnits(long duration) {
+            this.duration = duration;
+        }
+
+        public long getDuration() {
+            return duration;
+        }
+    }
+
+    /**
+     * Creates a {@link J4pClient} for the specified pod.
+     *
+     * @param kubernetes The {@link KubernetesClient} instance.
+     * @param pod        The name of the pod.
+     * @return An instance of the {@link J4pClient}.
+     */
+    private static J4pClient forPod(KubernetesClient kubernetes, String pod) {
+        String jolokiaUrl = String.format(JOLOKIA_URL_FORMAT, kubernetes.getMasterUrl(), kubernetes.getNamespace(), pod);
+        try {
+            return new J4pClientBuilder()
+                .url(jolokiaUrl)
+                .user("user")
```

Comment:
I also would set "POST" as the default method, as its much more robust then "GET" (because of escaping issuees). Actually this should be a default for Jolokia but unfortunately it is not.